NMCA_001/NMCA_001_script.lua

```diff
@@ -0,0 +1,767 @@
+    end
+
+    ----------
+    -- Spawn PBase
+    ----------
+    local PBase = ScenarioUtils.CreateArmyGroup('UEF', 'PowerBase')
+    local PBaseWalls = ScenarioUtils.CreateArmyGroup('UEF', 'M3PWalls')
+    ScenarioInfo.Shields = ScenarioUtils.CreateArmyGroup('UEF', 'M3Shields')
+
+    ----------
+    -- Spawn Patrols
+    ----------
+    local units = ScenarioUtils.CreateArmyGroupAsPlatoon('UEF', 'M3_AirPatrol_' .. Difficulty, 'NoFormation')
+
+    for k, v in units:GetPlatoonUnits() do
+        ScenarioFramework.GroupPatrolRoute({v}, ScenarioPlatoonAI.GetRandomPatrolRoute(ScenarioUtils.ChainToPositions('M3_AirPatrol_Chain')))
+    end
+
+    local AttackUnits = ScenarioUtils.CreateArmyGroupAsPlatoon('UEF', 'M3_Init_Attack_' .. Difficulty, 'AttackFormation')
+
+    for k, v in AttackUnits:GetPlatoonUnits() do
+        ScenarioFramework.GroupPatrolRoute({v}, ScenarioPlatoonAI.GetRandomPatrolRoute(ScenarioUtils.ChainToPositions('M3_LandAttack_Chain')))
+    end
+
+    local PBasePatrol = ScenarioUtils.CreateArmyGroupAsPlatoon('UEF', 'M3PBasePatrol', 'NoFormation')
+
+    for k, v in PBasePatrol:GetPlatoonUnits() do
+        ScenarioFramework.GroupPatrolRoute({v}, ScenarioPlatoonAI.GetRandomPatrolRoute(ScenarioUtils.ChainToPositions('M3_PBase_Patrol_Chain')))
+    end
+
+    ----------
+    -- Dialogue
+    ----------
+    ScenarioFramework.Dialogue(OpStrings.M2_Update_Commander, M2ObjectviesExtended, true)
+end
+
+function M2ObjectviesExtended()
+    ----------
+    -- Objectives
+    ----------
+    ScenarioInfo.M2P3 = Objectives.Capture(
+    'primary',                      -- type
+    'incomplete',                   -- complete
+    'Capture Enemy Tech Centre',  -- title
+    'We need to find out what is going on. Capture the enemy tech centre so we can access the data.',  -- description
+    {
+        MarkUnits = true,
+        Units = {ScenarioInfo.UEFTech},
+    })
+    table.insert(AssignedObjectives, ScenarioInfo.M2P3)
+
+    ScenarioInfo.M2S1 = Objectives.Capture(
+    'secondary',                      -- type
+    'incomplete',                   -- complete
+    'Capture Enemy Factory',  -- title
+    'Our Tech is very weak compared to the enemy. Capture one of their factories to gain an advantage.',  -- description
+        {
+            MarkUnits = true,                               -- target
+            Units = {ScenarioInfo.UEFFactory},
+        }
+    )
+    table.insert(AssignedObjectives, ScenarioInfo.M2S1)
+
+    ScenarioInfo.M2P3:AddResultCallback(
+    function(result)
+        if(result) then
+            ScenarioFramework.Dialogue(OpStrings.M2TechCaptured, M3, true)
+        end
+    end)
+
+    ScenarioInfo.M2S1:AddResultCallback(
+    function(result, unit)
+        ScenarioInfo.UEFFactory = unit
+        FactoryCaptured = true
+        ScenarioFramework.Dialogue(OpStrings.UEFTauntFactory, nil, true)
+    end)
+end
+
+function M3()
+    ForkThread(function()
+        WaitSeconds(1)

```

Comment:
this is useless, it forks a function that wait one second as does nothing while it executes rest of the M3 funciton, remove please